pkg/components/rook/component.go

```diff
@@ -116,8 +117,20 @@ func (c *component) Metadata() components.Metadata {
 func convertNodeSelector(m map[string]string) string {
 	var ret string
 
-	for k, v := range m {
-		ret += fmt.Sprintf("%s=%s; ", k, v)
+	// covert map to slice and sort them
+	// by keys.
+	keys := make([]string, len(m))
+	i := 0
+
+	for k := range m {
+		keys[i] = k
+		i++
+	}
+
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		ret += fmt.Sprintf("%s=%s; ", k, m[k])
```

Comment:
could you write unit tests for this function?

Reply:
I checked with #938 it worked, and I think unit tests should be added, once 938 is merged.

Reply:
Also to make sure https://play.golang.org/p/eeSiutD3Z1D

Reply:
Yeah, I'd add unit test for it.

Reply:
cool, adding